File: tiq-e/array/7/main.ts
import {performance} from 'perf_hooks';

function plusOne(digits: number[]): number[] {
  for (let i = digits.length - 1; i >= 0; i--) {
    if (digits[i] < 9) {
      digits[i] += 1
      return digits
    }

    digits[i] = 0
  }
  digits.unshift(1)
  return digits
};

let arr = [3, 1, 2, 1, 2];

let start = performance.now();
let result = plusOne(arr);
let end = performance.now();
console.log('Duration:', end - start);
console.log('Result:', result)

File: tiq-e/array/7/main.go
package main

import (
	"fmt"
	"time"
)

func plusOne(digits []int) []int {
	carry := 1

	for i := len(digits) - 1; i >= 0; i-- {
		digits[i], carry = (digits[i]+carry)%10, (digits[i]+carry)/10
		println(carry)
	}

	if carry != 0 {
		return append([]int{carry}, digits...)
	}

	return digits
}

func main() {
	arr := []int{3, 2, 2, 6, 4}

	var start = time.Now()
	var result = plusOne(arr)
	var end = time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}
